### DISCUSSION TITLE
Prefix Enum-references with Enum::

### DISCUSSION THREAD

**Discussion by rvanbekkum** [StefanMaron/BusinessCentral.LinterCop#66] [New Rule]
It would be nice to have a rule that checks that enum-references are prefixed with `Enum::`.
So instead of typing `"Comment Line Table Name"::Customer` type `Enum::"Comment Line Table Name"::Customer`.
Or, when one of the built-in methods are used, e.g., `Enum::"Comment Line Table Name".FromInteger(myInt)`.

**Comment by fvet** [comment]
Please do not enable by default. I tend not to write **Enum::** ;)

**Reply by rvanbekkum** [reply]
Why the thumbs down though? 😅 I started doing this as the IntelliSense works better with this and it is also more clearly shows what you are referencing, i.e., directly referencing the enum rather than referencing a field, variable, or something else. I think it's a good practice to add it rather than omit it.

**Reply by StefanMaron** [reply]
I think this would be a candidate for warning be default 🤔🤷‍♂️
